src/Forms/TurnstileCaptchaField.php

```diff
@@ -172,71 +175,57 @@ public function validate($validator)
+        $curlOptions = [];
+        $secret_key = $this->_secretKey ?: Environment::getEnv('SS_TURNSTILE_SECRET_KEY');
         $proxy_server = $this->_proxyServer ?: self::config()->proxy_server;
         if (!empty($proxy_server)) {
-            curl_setopt($ch, CURLOPT_PROXY, $proxy_server);
+            $curlOptions[CURLOPT_PROXY] = $proxy_server;
 
             $proxy_auth = $this->_proxyAuth ?: self::config()->proxy_auth;
             if (!empty($proxy_auth)) {
-                curl_setopt($ch, CURLOPT_PROXYUSERPWD, $proxy_auth);
+                $curlOptions[CURLOPT_PROXYUSERPWD] = $proxy_auth;
             }
 
             $proxy_port = $this->_proxyPort ?: self::config()->proxy_port;
             if (!empty($proxy_port)) {
-                curl_setopt($ch, CURLOPT_PROXYPORT, $proxy_port);
+                $curlOptions[CURLOPT_PROXYPORT] = $proxy_port;
             }
         }
-
-        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, self::config()->verify_ssl);
-        curl_setopt($ch, CURLOPT_POST, true);
-        curl_setopt($ch, CURLOPT_USERAGENT, str_replace(',', '/', 'SilverStripe'));
-        curl_setopt(
-            $ch,
-            CURLOPT_POSTFIELDS,
-            http_build_query([
-                'secret' => $secret_key,
-                'response' => $recaptchaResponse,
-                'remoteip' => $request->getIP(),
-            ])
-        );
-        $response = json_decode(curl_exec($ch), true);
-
-        if (is_array($response)) {
-            $this->verifyResponse = $response;
-
-            if (!array_key_exists('success', $response) || !$response['success']) {
-                $validator->validationError($this->name, _t(
-                    'Terraformers\\TurnstileCaptcha\\Forms\\CaptchaField.EMPTY',
-                    '_Please answer the captcha,
-                     if you do not see the captcha you must enable JavaScript'),
-                    'validation');
-                return false;
-            }
-
-        } else {
+        $curlOptions[CURLOPT_RETURNTRANSFER] = true;
+        $curlOptions[CURLOPT_SSL_VERIFYPEER] = self::config()->verify_ssl;
+        $curlOptions[CURLOPT_USERAGENT] = str_replace(',', '/', 'SilverStripe');
+        $client = HttpClient::create()->getClient();
+
+       try {
+           $response = $client->request('POST', 'https://challenges.cloudflare.com/turnstile/v0/siteverify',[
+               'json' => [
+                   'secret' => $secret_key,
+                   'response' => $recaptchaResponse,
+                   'remoteip' => $request->getIP(),
+               ],
+               'curl' => $curlOptions,
+               'timeout' =>10
+           ]);
+       }catch (Throwable $e) {
+           if($e instanceof RequestException && $e->hasResponse()){
+               return  $e->getResponse();
+           }
+       }
+
+        if ($response->getStatusCode() !== 200) {
```

Comment:
the docs say to check the 'outcome' property of the response so I wouldn't rely on just the response code. They could use that to indicate you successfully submitted a response but with a failed 'outcome'

Reply:
see https://developers.cloudflare.com/turnstile/get-started/server-side-validation/

Reply:
eg if I decode the body of the response when I use an invalid key I get:

![image](https://github.com/silverstripe-terraformers/turnstile-captcha/assets/948122/7e328f1e-b33e-4f3a-828c-a8b83b19b571)